### ISSUE TITLE
Deploy with Meteor Up causes crash when uploading

### ISSUE THREAD

**Issue by AdamSutch** [vsivsi/meteor-file-collection#33]
Hi,

I'm not sure if this is a problem with Meteor Up or this package.
I've tested with the meteor-file-sample-app and it works when running it locally on Mac (10.10.2) and Ubuntu (14.04) but when I deploy with Meteor Up it crashes during file upload (download is fine).

I'm not getting an error message when the app crashes but upon restart the upload finishes.

To reproduce:
1. Download meteor-file-sample-app
2. Mup init
3. Setup config for server (using Ubuntu on Digital Ocean)
4. Mup setup
5. Mup deploy
6. Try to upload any file - bigger causes more crashes

I'm not sure how to start debugging this as it works fine locally and doesn't log any errors on the server. (Outside of Meteor Up saying the app is restarting)

Any help would be much appreciated,
Adam

**Comment by vsivsi** [comment]
Hi, sorry for the delay in responding. I'll look into this tomorrow.

**Comment by AdamSutch** [comment]
Thank you for your help,
If it helps I was able to get somewhat of an error message, however I don't think error code "null" is anything useful.

error: Forever detected script exited with code: null
error: Script restart attempt #1
 >> stepping down to gid: meteoruser
 >> stepping down to uid: meteoruser

**Comment by vsivsi** [comment]
Hi, maybe a silly question, but how are you able to verify that file download works when using Meteor Up if it crashes whenever you upload?  Does upload work under some circumstances but not others?

**Comment by AdamSutch** [comment]
It will upload a few chunks then the app crashes (forever restarts it) and a few other chunks are uploaded. This repeats until the upload finishes, at that point I'm able to download (or stream) it just fine.
I'm guessing it's resumable.js that allows it to finish.

It isn't consistent on the number of chunks it's able to upload before crashing. Sometimes it 1 or 2 other times it's all but the last few.

It seems ok with small single chunk files.

I've tested on our staging server with both my video streaming site and with your sample app. (I thought it might have been something in my code)

The server has 512MB ram but I've tested it with a VirtualBox Ubuntu install with 8GB and the same thing happens.

**Comment by vsivsi** [comment]
Okay. What versions of everything are you using... Meteor, this package, mongodb, etc?

**Comment by vsivsi** [comment]
Also, to help narrow down what is happening, can you try enabling and uploading files using straight POST and/or PUT HTTP calls, as shown in the `curl` examples here: https://github.com/vsivsi/meteor-file-collection#http-request-behaviors

If it works with curl + POST and curl + PUT, then we'll know that this is something specific to using resumable.js, and not some more general issue with uploading/writing files in this environment.

**Comment by AdamSutch** [comment]
For simplicity I explain the meteor-file-sample-app

Meteor: 1.0.3.2
meteor-file-sample-app: commit 57275dab3e2046f403cf0e837b1a8c3bc15ec061
Ubuntu: 14.04 (up to date)
MongoDB: 2.6.8
Node: v0.10.33
Meteor-up: 0.9.4
Deploying from Mac osx: 10.10.2

I'll test the above suggest in a minute.

**Comment by AdamSutch** [comment]
The commands seem to work. I also tried updating the latest of everything and it seems I'm able to upload more chunks before it crashes (14 or 15). Files less than ~15MB seem to ok now (it was >1MB before)

**Comment by vsivsi** [comment]
Hi, I haven't really used Mup before. Is it possible for you to ssh into your DigOcean instance and start the app manually, so you can see everything that the server writes to the console?  It is very strange that it is crashing without a stack trace or anything, and that makes it hard to know what is going on. 

If necessary, I'll set up a Virtualbox Ubuntu VM and try this out myself, but if you can get any additional log info out of your server, that would really help.

**Comment by vsivsi** [comment]
@AdamSutch   I just tried to replicate this issue using a local VBox Ubuntu 14.04 Server install.

I'm using mup 0.9.6, and the latest of everything else (including a few changes I just pushed to the sample app). I've never used mup before, but everything was pretty easy.

On the first try, I was able to successfully upload 3 files totaling almost 2GB to the sample app via resumable.js / drag and drop, without any errors. 

So as of right now, I guess I can't replicate this.

**Comment by AdamSutch** [comment]
Yes, after updating everything (including your latest changes) I haven't been able to reproduce on the sample app.
I'm not sure what exactly fixed it, but it's working now.

Thank you for your time

**Comment by AdamSutch** [comment]
Just incase anyone else runs into this problem. I found the cause of the issue.
Using the sample app it worked because it didn't require much ram.
When I was running my site, it had lots of additional collections beyond the files.
What was happening was Mongo and/or Node were running out of ram and swap isn't enabled by default on Digital Ocean Ubuntu instances. When I added swap it fixed the crash.
